Bedrock/NSObject+Bedrock.h

```diff
@@ -8,21 +8,25 @@
 
 #import <Foundation/Foundation.h>
 
+@protocol BRNSObjectDescribable <NSObject>
+@end
+
+@interface NSString (BRNSObjectDescribable) <BRNSObjectDescribable>
+@end
+
+@interface NSDictionary (BRNSObjectDescribable) <BRNSObjectDescribable>
+@end
+
 @interface NSObject (Bedrock)
 
-// Every object in <keys> must be either an NSString, or a single-entry NSDictionary<NSString *, id>.
-//
-// NSString entries will be treated as the key in a key-value pair. KVC will be used to find a value,
-// and if the value is non-nil, it will be added to the resulting description string.
+// Property Keys is an array of NSString and NSDictionary objects. An NSString
+// object is interpreted as both a label and a KVC key, whereas a dictionary (which
+// should contain two NSStrings), is interpreted as a {label: value} pair.
 //
-// NSDictionary<NSString *, id> entries will be treated as a key-value pair. The dictionary key will be
-// treated as a key, and the the result of [anObject description] will be used for the object.
-//
-- (NSString *)br_descriptionWithKeys:(NSArray *)keys;
-
+// e.g.:	return [self ap_descriptionForPropertyKeys:@[@"name", @{@"age": self.calculateAge}]];
+- (NSString *)br_descriptionForPropertyKeys:(NSArray<id<BRNSObjectDescribable>> *)propertyKeys;
 
-// This is used in the same way as -br_descriptionWithKeys:, but the object pointer isn't included
-// in the string. This can be useful for value types or immutable objects, where the reference may not matter.
-- (NSString *)br_structDescriptionWithKeys:(NSArray *)keys;
+// As above, but without the object pointer info
+- (NSString *)br_structDescriptionForPropertyKeys:(NSArray<id<BRNSObjectDescribable>> *)propertyKeys;
```

Comment:
As discussed on a call, I don't think we should modify this category (both header and implementation) at all:
1. Changing method names/signatures will lead to breakage in other existing users.
2. `BRNSObjectDescribable` protocol seems to be used just as a "marker", and doesn't add much value.
3. Implementation of `br_descriptionForPropertyKeys:descriptionStyle:` is almost identical, but actually _loses_ functionality.
4. `DescriptionStyleObjectExpanded` case is effectively unused.
5. `br_prettyClassName` is not exposed in the header and is not used internally.

Reply:
Reverted

Reply:
@vlas-voloshin Just confirming that I should leave this header & implementation completely unchanged, and continue using NSDictionary+APExtensions in The O?

Reply:
(I assume you meant NSObject+Bedrock, not NSDictionary+APExtensions?)
Current implementation in Bedrock is _more_ flexible than what version in The O is. My suggestion was to keep Bedrock implementation intact, remove the corresponding implementation in The O, and move its clients to the corresponding Bedrock methods.